st2common/st2common/services/triggers.py

```diff
@@ -169,8 +159,16 @@ def create_or_update_trigger_db(trigger):
 
 
 def create_trigger_db_from_rule(rule):
-    trigger_api = _get_trigger_api_given_rule(rule)
-    return create_trigger_db(trigger_api)
+    trigger_dict = _get_trigger_dict_given_rule(rule)
+    existing_trigger_db = _get_trigger_db(trigger_dict)
+    # For simple triggertypes (triggertype with no parameters), we create a trigger when
+    # registering triggertype. So if we hit the case that there is no trigger in db but
+    # parameters is empty, then this case is a run time error.
+    if not trigger_dict.get('parameters', {}) and not existing_trigger_db:
+        raise Exception('A simple trigger should have been created when registering '
```

Comment:
Yes, I had the same thought. I was just lazy to add a custom exception. Throwing ValueError seems less ideal. Let me go the custom exception route.